portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,17 +34,59 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+        // Create Comment from request.
+        Comment c = Comment.makeComment(request);
+
+        // Set properties in Datastore entity.
+        Entity commentEntity = new Entity("comment");
+        commentEntity.setProperty("timestamp", c.getTimestamp());
+        commentEntity.setProperty("author", c.getAuthor());
+        commentEntity.setProperty("text", c.getText());
+        datastore.put(commentEntity);
+
+        // Return the comment ID.
+        Gson gson = new Gson();
+        response.setContentType("text/plain");
+        response.getWriter().println(commentEntity.getKey().getId());
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+        // Get comment limit parameter.
+        int max = Integer.parseInt(request.getParameter("max"));
+
+        // Get page number.
+        int page;
+        try {
+            page = Integer.parseInt(request.getParameter("page"));
+        } catch (NumberFormatException e) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return;
+        }
+        
+        ArrayList<Comment> comments = new ArrayList<>();
+
+        // Create a query for the Comments.
+        Query query = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
+        PreparedQuery results = datastore.prepare(query);
+
+        // Skip earlier pages and enforce Comment limit.
+        FetchOptions options = FetchOptions.Builder.withLimit(max);
+        options.offset((page - 1) * max);
```

Comment:
Can you add a comment describing why we're doing this?

Reply:
Added a description.